api/envoy/api/v2/core/config_source.proto

```diff
@@ -102,8 +102,19 @@ message RateLimitSettings {
 // <arch_overview_service_discovery>` etc. may either be sourced from the
 // filesystem or from an xDS API source. Filesystem configs are watched with
 // inotify for updates.
-// [#next-free-field: 6]
+// [#next-free-field: 7]
 message ConfigSource {
+  enum XdsApiVersion {
+    // set api version automatically, default version is set as v2.
```

Comment:
I'm wondering whether this should be match the version of `ConfigSource`, i.e.  if the type is v3alpha.ConfigSource, use v3alpha. WDYT @htuch?

Reply:
Making it explicit (as done in this PR) has the advantage that we can support mixed v2/v3 configurations, e.g. LDS loads from a v3 server but RDS comes from v2. We may get into a situation where a v3 Listener one day is expecting to have some v3-related route table delivered, but I think we can enforce these through consistency checking at config ingestion, the general case is that we support mixed.

Reply:
To be clear, I meant this for `AUTO`, we should definitely keep the rest of options explicit.

Reply:
Yeah, I think that would be reasonable semantics.

Reply:
I think let's just say that it's V2 for now, it would be ideal to have folks specifically exactly what they are after going forward, and this will encourage that.